tests/tests/test_filetransfer.py

```diff
@@ -281,4 +323,406 @@ def test_filetransfer(self, enterprise_no_client):
+        time.sleep(4)  # wait for mender-connect re-connect
+
+        logger.info(
+            "-- testcase: File Transfer limits: file inside chroot; upload allowed"
+        )
+        f = NamedTemporaryFile(delete=False)
+        f.write(os.urandom(16))
+        f.close()
+        # upload a file
+        r = upload_file(
+            "/var/lib/mender/filetransfer/random.bin",
+            open(f.name, "rb"),
+            devid,
+            authtoken,
+        )
+
+        assert r.status_code == 201
+
+        set_limits(
+            mender_device,
+            {
+                "Enabled": True,
+                "FileTransfer": {"MaxFileSize": 16378, "FollowSymLinks": True,},
+            },
+        )
+        time.sleep(4)  # wait for mender-connect re-connect
+
+        logger.info(
+            "-- testcase: File Transfer limits: file size over the limit; upload forbidden"
+        )
+        f = NamedTemporaryFile(delete=False)
+        for i in range(128 * 1024):
+            f.write(b"ok")
+        f.close()
+        r = upload_file("/tmp/random.bin", open(f.name, "rb"), devid, authtoken,)
+
+        assert r.status_code == 400, r.json()
+        assert (
+            r.json().get("error")
+            == "failed to write file chunk: transmitted bytes limit exhausted"
+        )
+
+        set_limits(
+            mender_device,
+            {
+                "Enabled": True,
+                "FileTransfer": {
+                    "FollowSymLinks": True,
+                    "Counters": {"MaxBytesRxPerMinute": 16784,},
+                },
+            },
+        )
+        time.sleep(4)  # wait for mender-connect re-connect
+
+        logger.info(
+            "-- testcase: File Transfer limits: transfers during last minute over the limit; upload forbidden"
+        )
+        f = NamedTemporaryFile(delete=False)
+        for i in range(128 * 1024):
+            f.write(b"ok")
+        f.close()
+        upload_file(
+            "/tmp/random-0.bin", open(f.name, "rb"), devid, authtoken,
+        )
+        upload_file(
+            "/tmp/random-1.bin", open(f.name, "rb"), devid, authtoken,
+        )
+        logger.info("-- testcase: File Transfer limits: sleeping to gather the avg")
+
+        time.sleep(32)  # wait for mender-connect to calculate the 1m exp moving avg
+        mender_device.run(
+            "kill -USR1 `pidof mender-connect`"
+        )  # USR1 makes mender-connect print status
+
+        r = upload_file("/tmp/random-2.bin", open(f.name, "rb"), devid, authtoken,)
+
+        assert r.status_code == 400, r.json()
+        assert r.json().get("error") == "transmitted bytes limit exhausted"
+
+        logger.info(
+            "-- testcase: File Transfer limits: transfers during last minute: test_filetransfer_limits_upload sleeping 64s to be able to transfer again"
+        )
+        # let's rest some more and increase the limit and try again
+        time.sleep(64)  # wait for mender-connect re-connect
```

Comment:
Comment is wrong, you are not waiting for mender-connect to re-connect

Reply:
all sleeps: after the configuration file change, we have to restart mender-connect, all the sleeps are just to be sure mender-connect restarts and connects. The sleep is "the cheapest" way. it seemed wait_for_connect an over kill, but I can change that.

Reply:
changed as described in the above comment.

Reply:
(removed wrong comments here)
copy-paste on this line, but the above comment is right.
removed the one next to sleep.